src/later_posix.cpp

```diff
@@ -147,21 +149,40 @@ InputHandler* dummyInputHandlerHandle;
 static void remove_dummy_handler(void *data) {
   ASSERT_MAIN_THREAD()
   removeInputHandler(&R_InputHandlers, dummyInputHandlerHandle);
-  close(dummy_pipe_in);
-  close(dummy_pipe_out);
+  if (dummy_pipe_in  > 0) {
+    close(dummy_pipe_in);
+    dummy_pipe_in  = -1;
+  }
+  if (dummy_pipe_out > 0) {
+    close(dummy_pipe_out);
+    dummy_pipe_out = -1;
+  }
 }
 
-void deInitialize() {
+// Callback to run in child process after forking.
+void child_proc_after_fork() {
   ASSERT_MAIN_THREAD()
   if (initialized) {
     removeInputHandler(&R_InputHandlers, inputHandlerHandle);
-    close(pipe_in);
-    close(pipe_out);
-    initialized = 0;
 
-    // Trigger remove_dummy_handler()
-    // Store `ret` because otherwise it raises a significant warning.
-    ssize_t ret = write(dummy_pipe_in, "a", 1);
+    if (pipe_in  > 0) {
+      close(pipe_in);
+      pipe_in = -1;
+    }
+    if (pipe_out > 0) {
+      close(pipe_out);
+      pipe_out = -1;
+    }
+    if (dummy_pipe_in  > 0) {
+      close(dummy_pipe_in);
+      dummy_pipe_in  = -1;
+    }
+    if (dummy_pipe_out > 0) {
+      close(dummy_pipe_out);
+      dummy_pipe_out = -1;
+    }
+
+    initialized = 0;
```

Comment:
Do you also need to remove the `dummyInputHandlerHandle`? I feel like we talked about this but I can't remember

Reply:
I think it can stay. (We did talk about this.) Removing it is tidier, but not strictly necessary because it doesn't hurt anything in the child (or parent) process.